### PR TITLE
fixed the false time error when user specifies the new instance name

### LINKED ISSUE #2772: False error: timeout on start VM from template
**Describe the bug**
When start a VM from a template multipass exits with non zero telling that it has timed out.
However, template timeout is 1200s (20 min). Real execution process was 486s

**To Reproduce**
1. `multipass launch charm-dev -n name-vm -m 8g -c 3 -d 30G`

Get:

When shell to the machine:

**Expected behavior**
No timeout error, normal start

**Additional info**
 - OS: 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"

- `multipass version`
multipass   1.10.1
multipassd  1.10.1

- `multipass info --all`
```
Name:           charm-dev
State:          Running
IPv4:           10.112.13.105
                10.135.56.1
                10.1.157.64
Release:        Ubuntu 20.04.4 LTS
Image hash:     77f2f05e39e5 (Ubuntu 20.04 LTS)
Load:           0.78 0.85 1.01
Disk usage:     15.7G out of 19.2G
Memory usage:   2.3G out of 7.8G
Mounts:         /home/maksim/git/fastapi-demo => /fastapi-demo
                    UID map: 1000:default
                    GID map: 1000:default

Name:           operator-day
State:          Stopped
IPv4:           --
Release:        --
Image hash:     77f2f05e39e5 (Ubuntu 20.04 LTS)
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         --

Name:           prometheus-vm
State:          Running
IPv4:           10.112.13.235
                10.66.176.1
                10.1.91.192
Release:        Ubuntu 22.04.1 LTS
Image hash:     0fcb96fbfd71 (Ubuntu 22.04 LTS)
Load:           0.46 0.55 0.37
Disk usage:     10.3G out of 28.9G
Memory usage:   1.7G out of 7.8G
Mounts:         /home/maksim/git/prometheus-k8s-operator => /prometheus-k8s-operator
                    UID map: 1000:default
                    GID map: 1000:default

Name:           vivid-kakapo
State:          Stopped
IPv4:           --
Release:        --
Image hash:     77f2f05e39e5 (Ubuntu 20.04 LTS)
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         --

Name:           zinc-vm-bare
State:          Stopped
IPv4:           --
Release:        --
Image hash:     77f2f05e39e5 (Ubuntu 20.04 LTS)
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         --
```

- `multipass get local.driver`
qemu

**Logs**
[logs.txt](https://github.com/canonical/multipass/files/9716090/logs.txt)

### PR DESCRIPTION
**Pull request by georgeliao**
Fixes #2772

### CODE REVIEW

**Review by townsend2010 — CHANGES_REQUESTED** [review]
Hey @georgeliao!

Thanks for this!  It fixes the issue at hand as far as I can tell.

I'd like to make a few general comments while you are learning the workflow of making changes in Multipass.

I see a number of changes that aren't really related to the fix itself- we tend to call these "drive by" changes and as reviewers, it makes it more difficult to tell what is fixing the actual issue vs. things you are just changing.  For example, changes in variable names, variable types, return types, and function names to name a few.

If a PR is fixing a bug, the PR should be focused on that.  If you want to propose changes to those types of things I mentioned previously, a separate PR is greatly appreciated with a description saying this is for renaming variables, changing types, etc.

That said, if it's a one or two line "drive by" fix in a separate commit in the same PR, we are usually fine with that and it doesn't add a bunch of extra lines in the diff. 

This way, it's clear to the reviewer what the intent of the PR is for.  I hope this makes sense.

For this PR, I prefer you revert the name changes back to what they were originally.  I don't mind you creating a follow up with the changes and at that point, we can discuss as a team if we would like those changes or a subset of the changes.

Again, thank you for the fix and putting this out there :slightly_smiling_face:

**Comment by georgeliao** [comment]
@townsend2010 
I see, thanks for letting me know. I will revert things back to bug-fix changes only.

**Comment by georgeliao** [comment]
Hi @townsend2010 

The requested changes have been done. The diff now is much smaller and purely focus on the bug fix.

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -2365,8 +2365,8 @@ void mp::Daemon::create_vm(const CreateRequest* request,
 
     // TODO: We should only need to query the Blueprint Provider once for all info, so this (and timeout below) will
     //       need a refactoring to do so.
-    auto name = name_from(checked_args.instance_name, config->blueprint_provider->name_from_blueprint(request->image()),
-                          *config->name_generator, vm_instances);
+    const std::string blueprint_name = config->blueprint_provider->name_from_blueprint(request->image());
```
Comment (townsend2010) [inline src/daemon/daemon.cpp:2368]: I'd prefer `const auto` here to follow the form of using `auto` where we can.

**Review by townsend2010 — APPROVED** [review]
Hey @georgeliao,

This looks great!  One little niggle inline below, but I certainly won't block on it.  I'm approving and you can decide if you want to change it.

**Comment by townsend2010** [comment]
bors delegate

**Comment by townsend2010** [comment]
bors delegate+

**Pull request merged**